SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java

```diff
@@ -171,12 +171,13 @@ private static void clearCurrentState() {
 
     private static void addWorldObjectToWorlds(HashMap<String, JSONObject> worlds, JSONObject worldJSON) throws JSONException {
         String worldId = worldJSON.getString("itemId");
-        worlds.put(worldId, worldJSON);
+        HashMap<String, JSONObject> worldsInternal = new HashMap<String, JSONObject>(worlds);
```

Comment:
@vedi, it's a hint given by @borisspektor, crash can cause because recursive calls here.